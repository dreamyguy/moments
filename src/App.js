import React, { Component } from 'react';
import './App.scss';
import Header from './components/layout/Header';
import Footer from './components/layout/Footer';

class App extends Component {
  render() {
    return (
      <div className="app">
        <Header/>
        <div className="content">
          {this.props.children}
        </div>
        <Footer/>
      </div>
    );
  }
}

export default App;
