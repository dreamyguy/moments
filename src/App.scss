$bottom-spacing: 4rem;

// Generic
.app {
  text-align: center;
  background-color: #222;
  min-height: 100vh;
  font-size: calc(10px + 2vmin);
  color: white;
}

// Layout
header,
footer {
  display: flex;
  justify-content: center;
  padding: 1rem;
  width: 100%;
}
footer {
  background-color: #111;
  padding: $bottom-spacing/2 0;
}
.content {
  max-width: 1280;
  padding-bottom: $bottom-spacing;
}

// Typography
.link {
  color: #61dafb;
}

// Helpers
.hidden {
  display: none;
}
