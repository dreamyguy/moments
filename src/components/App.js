import React, { Component } from 'react';
import Header from './layout/Header';
// import Footer from './layout/Footer';

class App extends Component {
  render() {
    return (
      <div className="app">
        <Header/>
        <div className="content">
          {this.props.children}
        </div>
        {/* <Footer/> */}
      </div>
    );
  }
}

export default App;
