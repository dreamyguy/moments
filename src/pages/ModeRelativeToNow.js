// Import dependencies
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// Import components
import DateCard from './../components/time/DateCard';
import DatePickerBaseDate from './../components/time/DatePickerBaseDate';
import TimeDifference from './../components/time/TimeDifference';

class ModeRelativeToNow extends Component {
  render () {
    const {
      baseDate,
      baseDateIsInThePast,
      mode,
      nowDate
    } = this.props;
    let classesFrstRow = 'col-12 col-md-6';
    let classesSecondRow = 'col-12 col-md-6';
    if (baseDateIsInThePast) {
      classesFrstRow = 'col-12 col-md-6 order-2';
      classesSecondRow = 'col-12 col-md-6 order-1';
    }
    return (
      <section className="section section--alt">
        <div className="container m-t-20 m-b-20">
          <>
            {mode &&
              <div className="row">
                <div className="col-12">
                  <DatePickerBaseDate/>
                </div>
              </div>
            }
            {mode && baseDate &&
              <>
                <div className="row">
                  <div className={classesFrstRow}>
                    <DateCard
                      heading = "Now"
                      date = {nowDate}
                    />
                  </div>
                  <div className={classesSecondRow}>
                    <DateCard
                      heading = "Set date"
                      date = {baseDate}
                      classes = "bl-fav-orange-dark"
                    />
                  </div>
                </div>
                <div className="row">
                  <div className="col-12">
                    <TimeDifference
                      timeA = {baseDate}
                      timeB = {nowDate}
                    />
                  </div>
                </div>
              </>
            }
          </>
        </div>
      </section>
    );
  }
}

const mapStateToProps = ({main}) => {
  return {
    baseDate: main.baseDate,
    baseDateIsInThePast: main.baseDateIsInThePast,
    mode: main.mode,
    nowDate: main.nowDate
  }
}

const ModeRelativeToNowConnect = connect(
  mapStateToProps
)(ModeRelativeToNow);

ModeRelativeToNow.propTypes = {
  baseDate: PropTypes.string,
  baseDateIsInThePast: PropTypes.bool,
  mode: PropTypes.string,
  nowDate: PropTypes.string
};

export default ModeRelativeToNowConnect;
