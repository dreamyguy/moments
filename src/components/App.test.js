// Import dependencies
import React from 'react';
import {shallow} from 'enzyme';
import {Provider} from 'react-redux';

// Import components
import App from './App';

// Import a page component, just for testing the rendering
import ModeRelativeToNow from './../pages/ModeRelativeToNow';

// Import store
import initStore from './../store/store';

// Mock window
const mockWindow = {};

const store = initStore(mockWindow);

describe('App', () => {
  const app = shallow(
    <Provider store={store}>
      <App>
        <ModeRelativeToNow/>
      </App>
    </Provider>
  );
  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });
});
