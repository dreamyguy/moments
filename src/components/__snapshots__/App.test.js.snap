// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `ShallowWrapper {}`;
